zerver/tests/test_invite.py

```diff
@@ -850,26 +851,33 @@ def test_successful_invite_users_with_specified_streams(self) -> None:
         self.submit_reg_form_for_user(invitee, "password")
         self.check_user_subscribed_only_to_streams("test1", [denmark, sandbox, verona, zulip])
 
-    def test_can_invite_others_to_realm(self) -> None:
-        def validation_func(user_profile: UserProfile) -> bool:
-            return user_profile.can_invite_users_by_email()
-
-        realm = get_realm("zulip")
-        do_set_realm_property(
-            realm, "invite_to_realm_policy", InviteToRealmPolicyEnum.NOBODY, acting_user=None
-        )
-        desdemona = self.example_user("desdemona")
-        self.assertFalse(validation_func(desdemona))
-
-        self.check_has_permission_policies("invite_to_realm_policy", validation_func)
-
     def test_invite_others_to_realm_setting(self) -> None:
         """
-        The invite_to_realm_policy realm setting works properly.
+        The `can_invite_users_group` realm setting works properly.
         """
         realm = get_realm("zulip")
-        do_set_realm_property(
-            realm, "invite_to_realm_policy", InviteToRealmPolicyEnum.NOBODY, acting_user=None
+
+        administrators_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.ADMINISTRATORS, realm=realm, is_system_group=True
+        )
+        moderators_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MODERATORS, realm=realm, is_system_group=True
+        )
+        full_members_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.FULL_MEMBERS, realm=realm, is_system_group=True
+        )
+        members_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MEMBERS, realm=realm, is_system_group=True
+        )
+        nobody_system_group = NamedUserGroup.objects.get(
```

Comment:
We currently do not test for all the system user groups for other settings, but this is fine as well.